exo/api/chatgpt_api.py

```diff
@@ -31,65 +31,84 @@
     },
-    ### mistral
+    # mistral
     "mistral-nemo": {
         "MLXDynamicShardInferenceEngine": Shard(model_id="mlx-community/Mistral-Nemo-Instruct-2407-4bit", start_layer=0, end_layer=0, n_layers=40),
     },
     "mistral-large": {
         "MLXDynamicShardInferenceEngine": Shard(model_id="mlx-community/Mistral-Large-Instruct-2407-4bit", start_layer=0, end_layer=0, n_layers=88),
     },
+    # deepseek v2
+    "deepseek-coder-v2-lite": {
+        "MLXDynamicShardInferenceEngine": Shard(model_id="mlx-community/DeepSeek-Coder-V2-Lite-Instruct-4bit-mlx", start_layer=0, end_layer=0, n_layers=27),
+    }
 }
 
+
 class Message:
     def __init__(self, role: str, content: str):
         self.role = role
         self.content = content
 
+
 class ChatCompletionRequest:
     def __init__(self, model: str, messages: List[Message], temperature: float):
         self.model = model
         self.messages = messages
         self.temperature = temperature
 
+
 def resolve_tinygrad_tokenizer(model_id: str):
     if model_id == "llama3-8b-sfr":
         return AutoTokenizer.from_pretrained("TriAiExperiments/SFR-Iterative-DPO-LLaMA-3-8B-R")
     elif model_id == "llama3-70b-sfr":
         return AutoTokenizer.from_pretrained("TriAiExperiments/SFR-Iterative-DPO-LLaMA-3-8B-R")
     else:
-        raise ValueError(f"tinygrad doesnt currently support arbitrary model downloading. unsupported model: {model_id}")
+        raise ValueError(
+            f"tinygrad doesnt currently support arbitrary model downloading. unsupported model: {model_id}")
+
 
 async def resolve_tokenizer(model_id: str):
     try:
-        if DEBUG >= 2: print(f"Trying AutoTokenizer for {model_id}")
+        if DEBUG >= 2:
+            print(f"Trying AutoTokenizer for {model_id}")
         return AutoTokenizer.from_pretrained(model_id)
     except:
         import traceback
-        if DEBUG >= 2: print(traceback.format_exc())
-        if DEBUG >= 2: print(f"Failed to load tokenizer for {model_id}. Falling back to tinygrad tokenizer")
+        if DEBUG >= 2:
+            print(traceback.format_exc())
+        if DEBUG >= 2:
+            print(f"Failed to load tokenizer for {
+                  model_id}. Falling back to tinygrad tokenizer")
 
     try:
-        if DEBUG >= 2: print(f"Trying tinygrad tokenizer for {model_id}")
+        if DEBUG >= 2:
+            print(f"Trying tinygrad tokenizer for {model_id}")
         return resolve_tinygrad_tokenizer(model_id)
     except:
         import traceback
-        if DEBUG >= 2: print(traceback.format_exc())
-        if DEBUG >= 2: print(f"Failed again to load tokenizer for {model_id}. Falling back to mlx tokenizer")
+        if DEBUG >= 2:
+            print(traceback.format_exc())
+        if DEBUG >= 2:
+            print(f"Failed again to load tokenizer for {
+                  model_id}. Falling back to mlx tokenizer")
 
-    if DEBUG >= 2: print(f"Trying mlx tokenizer for {model_id}")
+    if DEBUG >= 2:
```

Comment:
Just a style thing, can you put these back on the same line

Reply:
@AlexCheema, would you share your linter or editconfig with me so I can set it up on my machine?

Reply:
Yup good point, see https://github.com/exo-explore/exo/pull/87#issuecomment-2254289002